### DISCUSSION TITLE
Need Help with Implementing GTM Server-Side Tagging in Next.js with @next/third-parties/google

### DISCUSSION THREAD

**Discussion by guybinya** [vercel/next.js#64562] [Help] [answered]
### Summary

I am currently working on a Next.js project and I'm looking to implement Google Tag Manager (GTM) with server-side tagging.
However, I am facing challenges in integrating server-side tagging with GTM into the Next.js environment with @next/third-parties/google GTM.
Is there any way to do it without using the <Script> tag?

**Comment by Icegreeen** [comment]
yea, in next14 you no longer need to use the `script`, see here in my article how you can integrate using @next/third-parties/google

https://icegreeen.medium.com/implementing-google-analytics-ga4-in-next-js-14-779da8df5c9e

**Reply by guybinya** [reply]
Thank you for your quick answer, still there is no way to edit the gtm source to match my server side tagging url

**Comment by 100tomtomas100** [comment]
I have implemented GTM as in your article and it dramaticaly slows down the initial page load. Are there any other ways of loading GTM after the page as now  for some reason it loads first?

**Reply by guybinya** [reply]
I agree about the performance, if you are using the pages router you can take a look on <Script strategy={"worker"} /> 

https://nextjs.org/docs/pages/api-reference/components/script#worker

**Reply by 100tomtomas100** [reply]
Im using app router :(

**Comment by guybinya** [comment, marked as answer]
Update: 
I made a PR adding the server side container support to @third-parties

https://github.com/vercel/next.js/pull/67161

**Reply by 100tomtomas100** [reply]
I saw it but it didnt get merged

**Reply by guybinya** [reply]
> I saw it but it didnt get merged

You can look at the diff and make your own GTM component.

**Reply by 100tomtomas100** [reply]
oh perfect 
Thank you ! :)

**Reply by doctacapital** [reply]
How can i find my serverSideContainerUrl? for my GoogleTagManager contianer?

**Reply by guybinya** [reply]
> How can i find my serverSideContainerUrl? for my GoogleTagManager contianer?

1. Go to google tag manager and select the server container. 
2. Go to the admin tab
![Screenshot 2025-01-14 at 16 19 18](https://github.com/user-attachments/assets/bd3565fc-92a5-48f3-85d5-bcfb79e2c362)
3. Click on "Container Settings"
![Screenshot 2025-01-14 at 16 19 24](https://github.com/user-attachments/assets/272d0f25-bd2d-4d14-b680-a7aa63397a09)
4. At the bottom you're going to see your default serverSideContainerUrl under "Default Url"

**Comment by doctacapital** [comment]
I cant find "Default Url"

Only have this values at Container Settings
<img width="961" alt="image" src="https://github.com/user-attachments/assets/e63ed974-7dbb-44d2-aae3-ad6a501bf319" />

**Comment by guybinya** [comment]
Your target platform should be server:
![Screenshot 2025-01-14 at 16 40 58](https://github.com/user-attachments/assets/d3685bd2-7d21-4c72-94d0-188434341fb9)

Try to create a new container and set it up as a server one.

**Comment by OpenInterYRZ** [comment]
Hi，I want to track the google login in my Next project ,the login is done on serverside.Now I have created a GCP server, and then what do I do?Do i need to fetch my GCP server? Or where's the tutorial for server side GTM...
